osmoutils/partialord/internal/dag/dag.go

```diff
@@ -0,0 +1,305 @@
+package dag
+
+import (
+	"fmt"
+	"sort"
+)
+
+// Dag struct maintains a directed acyclic graph, using adjacency lists to track edges.
+type Dag struct {
+	// there is a directed edge from u -> v, if directedEdgeList[u][v] = 1
+	// there is a directed edge from v -> u, if directedEdgeList[u][v] = -1
+	directedEdgeList []map[int]int8
```

Comment:
It can, but do you think that would be simpler? I found the negative relation simpler for groking direction

(Don't need to optimize for performance here, at relevant sizes its negligible)

Reply:
For the purpose of groking direction, agreed that negative relation would be more accurate, but considering code readability I'd agree with @p0mvn  to change this to boolean!

Reply:
How does code readability improve with bools, if the direction is simpler with negation?

Reply:
Although I had read the comment, I didn't build that link in my mind between the direction and the integer sign while reviewing. However, after you pointed it out, it makes perfect sense. It would probably be easier for me to review if I originally understood that link

I think representing binary values as booleans might feel more readable because if a developer reads one of the `Dag` methods and sees `directedEdgeList` map values used as `int8` before seeing this comment, it might confuse them at first. The reason for that is that they have to read the comment to understand that out of 2^8 possible values, only 2 are used.

It seems that both options have different pros and cons. @ValarDragon it's up to you which one to keep

Reply:
Going to keep negatives